source/upgrade.txt

```diff
@@ -0,0 +1,493 @@
+  for information the new ``setWindowFields()`` method signatures.
+
+  If your application uses this builder in a version earlier than v4.7, update
+  your source code to use the new method signature and rebuild your binary.
+
+.. _scala-breaking-changes-v4.2:
+
+Version 4.2 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This driver version introduces the following breaking changes:
+
+- The ``ObjectId`` class and its ``serialVersionUID`` field were updated
+  to use a new format that minimizes serialization compatibility issues
+  across different versions of the driver.
+
+  If an application using this version or later of the driver attempts to
+  perform Java Object Serialization on any objects that contain an
+  ``ObjectId`` and were serialized by a prior version of the driver, Java
+  throws an ``InvalidClassException``.
+
+  To learn more about Java Object Serialization, see the Java
+  Documentation on `Serializable Objects <https://docs.oracle.com/javase/tutorial/jndi/objects/serial.html>`__.
+
+.. _scala-breaking-changes-v4.0:
+
+Version 4.0 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This driver version introduces the following breaking changes:
+
+- Several classes and methods marked as deprecated in the 3.12 release
+  were removed in this version.
+- The insert helper methods return an insert result object instead of
+  ``void``.
+- The ``toJson()`` methods on ``BsonDocument``, ``Document``, and
+  ``DbObject`` return a relaxed JSON format instead of a strict JSON
+  format. This makes the JSON documents more readable, but can make it more
+  difficult to identify the BSON type information, such as the difference
+  between a 32-bit and 64-bit integer. If your application relies on the
+  strict JSON format, use the strict mode when reading or writing data.
+- The default BSON representation of ``java.util.UUID`` value was changed
+  from ``JAVA_LEGACY`` to ``UNSPECIFIED``. Applications that store or retrieve
+  UUID values must explicitly specify which representation to use. You can
+  specify the representation in the ``uuidRepresentation`` property of
+  ``MongoClientSettings``.
+
+  The UUID representation that you specify strictly controls how the driver
+  decodes UUIDs. In previous versions of the driver, if you specified the
+  ``JAVA_LEGACY`` representation, the driver would decode binary objects
+  of subtypes 3 and 4 as UUIDs. In version 4.0, the ``JAVA_LEGACY``
+  representation works only with subtype 3.
+
+  For a list of members in the ``UuidRepresentation`` enum, see the
+  `v4.0 API documentation <https://mongodb.github.io/mongo-java-driver/4.0/apidocs/bson/org/bson/UuidRepresentation.html>`__.
+
+- The connection pool no longer restricts the number of wait queue threads
+  or asynchronous tasks that require a connection to MongoDB. The
+  application will throttle requests as necessary rather than depend on
+  the driver to throw a ``MongoWaitQueueFullException``.
```

Comment:
```suggestion
  or asynchronous tasks that require a connection to MongoDB. The
  application throttles requests as necessary rather than depending on
  the driver to throw a ``MongoWaitQueueFullException``.
```